applications/vanilla/models/class.commentmodel.php

```diff
@@ -654,118 +654,113 @@ public function getUserScore($commentID, $userID) {
      * @param int $totalComments Total in entire discussion (hard limit).
      */
     public function setWatch($discussion, $limit, $offset, $totalComments) {
+        $userID = Gdn::session()->UserID;
+        if (!$userID) {
```

Comment:
I had a nagging suspicion that this wouldn't work out (because we tend to use the `$id > 0` checks all over the code base, but since out recent authenticator rewrite it seems pretty clear that we correct handle the weird case where a userID might be < 0 in our `Session->start()`).